### PR TITLE
Legacy configuration file support

### PR DESCRIPTION
**Pull request by forsthug**
See #9

### CODE REVIEW

File: src/fiskaltrust.Launcher/Commands/Common.cs [hunk 1]
```diff
@@ -24,13 +24,17 @@ public CommonCommand(string name) : base(name)
             AddOption(new Option<string?>("--log-folder"));
             AddOption(new Option<LogLevel?>("--log-level"));
             AddOption(new Option<string>("--launcher-configuration-file", getDefaultValue: () => "launcher.configuration.json"));
+            AddOption(new Option<string>("--legacy-config-file", getDefaultValue: () => "fiskaltrust.exe.config"));
```
Comment (volllly) [inline src/fiskaltrust.Launcher/Commands/Common.cs:27]: what do you think about the following?
reuse the `launcher-configuration-file` parameter and check if its file extension is `config` or `json`?

File: src/fiskaltrust.Launcher/Commands/Common.cs [hunk 2]
```diff
@@ -24,13 +24,17 @@ public CommonCommand(string name) : base(name)
...
+            AddOption(new Option<bool>("--use-legacy-config", getDefaultValue: () => false));
```
Comment (volllly) [inline src/fiskaltrust.Launcher/Commands/Common.cs:28]: And if the default value `launcher.configuration.json` is used and the file is not found we check if a `fiskaltrust.exe.config` exists and create a `launcher.configuration.json` with the migrated content.
Then we rename the `fiskaltrust.exe.config` to `fiskaltrust.exe.config.migrated`.
And maybe print a warning that we did that.

But that part is maybe to intransparent 🤷

File: src/fiskaltrust.Launcher/Configuration/LegacyConfigFileReader.cs [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            errors.Add((LogLevel.Information, $"Read legacy config file {path}.", null));
+            return launcherConfiguration;
+        }
+
+        private static void SetProperies(LauncherConfiguration launcherConfiguration, string? key, string? value)
+        {
+            if(key == "cashboxid")
+            {
+                launcherConfiguration.CashboxId = Guid.Parse(value);
+            }else if (key == "accesstoken")
```
Comment (volllly) [inline src/fiskaltrust.Launcher/Configuration/LegacyConfigFileReader.cs:42]:
```suggestion
            }
            else if (key == "accesstoken")
```

File: src/fiskaltrust.Launcher/Configuration/LegacyConfigFileReader.cs [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+                var appSettings = from item in purchaseOrder.Descendants("appSettings").DescendantsAndSelf("add")
+                                  select item;
+                
+                foreach (var item in appSettings)
+                {
+                    var key = item.Attribute("key")?.Value;
+                    var value = item.Attribute("value")?.Value;
+                    SetProperies(launcherConfiguration, key, value);
+                }
+            }catch(Exception e)
```
Comment (volllly) [inline src/fiskaltrust.Launcher/Configuration/LegacyConfigFileReader.cs:29]:
```suggestion
            }
            catch(Exception e)
```

File: src/fiskaltrust.Launcher/Commands/Common.cs (hunk 2 above)
Comment (volllly) [reply src/fiskaltrust.Launcher/Commands/Common.cs:28]: Or maybe even If a `fiskaltrust.exe.config` is found in the default path it is read and the `launcher.configuration.json` file is merged into it 
🤔

File: src/fiskaltrust.Launcher/Commands/Common.cs (hunk 1 above)
Comment (TSchmiedlechner) [reply src/fiskaltrust.Launcher/Commands/Common.cs:27]: I'd rather not do that, I think 🤔 see comment below

File: src/fiskaltrust.Launcher/Commands/Common.cs (hunk 2 above)
Comment (TSchmiedlechner) [reply src/fiskaltrust.Launcher/Commands/Common.cs:28]: And maybe call the parameter `--merge-legacy-config-if-exists`

**Review by TSchmiedlechner — APPROVED** [review]

File: src/fiskaltrust.Launcher/Configuration/LegacyConfigFileReader.cs [hunk 5]
```diff
@@ -0,0 +1,86 @@
+﻿using fiskaltrust.Launcher.Common.Configuration;
+using Newtonsoft.Json.Linq;
+using System.Xml.Linq;
+
+namespace fiskaltrust.Launcher.Configuration
+{
+    public class LegacyConfigFileReader
+    {
+        public static async Task<LauncherConfiguration> ReadLegacyConfigFile(List<(LogLevel logLevel, string message, Exception? e)> errors, string path)
```
[the remark this answers is not in the document]
Comment (volllly) [reply src/fiskaltrust.Launcher/Configuration/LegacyConfigFileReader.cs:9]: I have a PR ready to cleanup this whole pre config logging 😁 see #49

[the remark this answers is not in the document]
Comment (volllly) [reply src/fiskaltrust.Launcher/Configuration/LegacyConfigFileReader.cs:9]: After merging this one I will incorporate the changes in my pr

**Pull request merged**